pkg/config/metadata.go

```diff
@@ -127,3 +142,24 @@ func GetMetadataDefaults() (map[string]interface{}, map[string]interface{}) {
 func GetMetadataValueToPlaceholderPathsKeyMap() map[string][]string {
 	return mdValueToPlaceholderPathsKeyMap
 }
+
+// unmarshalToNestedStruct returns a struct with its nested values populated correctly
+func unmarshalToNestedStruct(originalValue interface{}, unmarshalToStruct interface{}) (interface{}, error) {
+	valAsJson, err := json.Marshal(originalValue)
+	if err != nil {
+		return nil, err
+	}
+
+	// all supported nested structs
+	switch unmarshalToStruct.(type) {
```

Comment:
it's seems a bit redundant to me to have `mdNestedValues = map[string]interface{}` to check whether the type is a valid nested struct and then do a switch/case on the type again. I feel like we could instead just have the switch/case, though that might be a little slower. 

Not firmly against how it is currently tho. Thoughts?

Reply:
- hmm I don't think they're doing the same thing though. `mdNestedValues` is just associating the key with its proper struct (not checking) while switch/case is doing the type validation
- can you clarify 'just have the switch/case' ?
  - I can't do a switch on v.(type) as that'll always return map[string]interface{}
  - I could do an explicit check for elasticInferenceAccel path instead of using mdNestedValues, but when we support more nested structs then I'd have to refactor to be more general